consensus/ibft/consensus_backend.go

```diff
@@ -0,0 +1,319 @@
+package ibft
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/consensus"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+func (i *backendIBFT) BuildProposal(blockNumber uint64) []byte {
+	var (
+		latestHeader      = i.blockchain.Header()
+		latestBlockNumber = latestHeader.Number
+	)
+
+	if latestBlockNumber+1 != blockNumber {
+		return nil
```

Comment:
I thought "Why not return error?". I think core ibft should handle error from this function even if core ibft doesn't do anything except for logging.

Reply:
@dbrajovic and I had lots of discussions on this topic.

The spec does not define a situation where you can insert a block, and error out, or create a block and error out.
It assumes that these things are handled by the `Backend` implementation itself - either the implementation panics and nothing is proposed, or it sends out a nil proposal - both situations are completely valid.

Why are they valid?

Because if the proposer doesn't propose anything, or the proposal is invalid (`nil`, for example), the timer that's ticking on all nodes will eventually expire and the nodes will go into a round change state, where they'll start the new round and someone else will get a chance to propose a valid block. The timer is what's preserving your liveness, and you can always, _always_, fall back on it.

The burden of the consensus package is to handle consensus, not corrupted DBs or the nodes inability to create a block (the `go-ibft` package has no notion or concept of a block, you can run it for an arbitrary byte array)

Reply:
On that note, I've added a log to make it clear something funky happened:

[e9595a8](https://github.com/0xPolygon/polygon-edge/pull/650/commits/e9595a8de32840c705c316cc4bb93331816c7947)